app/(auth)/uuid.ts

```diff
@@ -0,0 +1,25 @@
+import { cookies } from 'next/headers'
+import { NextResponse } from 'next/server'
+import type { Session } from '@supabase/supabase-js'
+
+const cookieName = 'user_uuid'
+
+export const getUuid = () => cookies().get(cookieName)?.value ?? undefined
+
+export const setUuidWithHeaders = (
+  headers: Headers,
+  session: Session
+): Headers => {
+  headers.append(
+    'set-cookie',
+    NextResponse.next()
+      .cookies.set(cookieName, session.user.id, {
+        httpOnly: true,
```

Comment:
Fixed little bit how we set cookies to prevent from those cookies set in previous position gone. 
Ref: https://github.com/vercel/next.js/discussions/48440